{{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/core/serializers.py

```diff
@@ -69,3 +75,24 @@ def validate(self, data):
 
     def create(self, validated_data):
         return User.objects.create_user(**validated_data)
+
+
+class ResetPasswordSerializer(serializers.Serializer):
+    code = serializers.CharField(allow_blank=False, required=True)
+    password = serializers.CharField(allow_blank=False, required=True)
+
+    def validate_code(self, value):
+        code_from_db = (
+            self.context.get("user")
+            .reset_password_codes.filter(created__gte=(timezone.now() - timedelta(minutes=settings.RESET_PASSWORD_CODE_VALIDITY_MINUTES)))
+            .first()
+        )
```

Comment:
shouldn't they only ever have a single code at any moment? If it expires, then it fails validation.
If they request a new one, the old one should be deleted.